src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/ActiveDirectoryAuthenticationProvider.cs

```diff
@@ -477,6 +518,30 @@ private static byte[] GetHash(string input)
             return hashedBytes;
         }
 
+        private static bool AreEqual(byte[] a1, byte[] a2)
+        {
+            if (a1 is null && a2 is null)
+            {
+                return true;
+            }
+            else if (a1 is null || a2 is null)
+            {
+                return false;
+            }
+            else if (a1.Length != a2.Length)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < a1.Length; i++)
+            {
+                if (a1[i] != a2[i])
+                    return false;
+            }
+
+            return true;
+        }
```

Comment:
I understand this is still a WIP, but I could not help myself from looking into optimizing that for loop. I think we can benefit from more modern C# features, since we are not supporting netstandard anymore. By using [ReferenceEquals](https://learn.microsoft.com/en-us/dotnet/api/system.object.referenceequals?view=net-8.0) and `Span<T>`, we can simplify and optimize the array comparison. This is just an untested comment, but I thought it would be good if we could implement it.

We could use `ReferenceEquals` to handle cases where both arrays are `null` or having same memory reference. This will simplify the code and make it more concise. At the end, our optimized implementation would resemble the following:
```C#
private static bool AreEqual(byte[] a1, byte[] a2)
{
    if (ReferenceEquals(a1, a2))
    {
        return true;
    }

    if (a1 is null || a2 is null)
    {
        return false;
    }

    if (a1.Length != a2.Length)
    {
        return false;
    }

    return a1.AsSpan().SequenceEqual(a2.AsSpan());
}
```